articles/test-runner-plugins.md

```diff
@@ -0,0 +1,233 @@
+
+```js
+define([
+    'taoTests/runner/plugin'
+], function(pluginFactory) {
+    'use strict';
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+
+        name: 'hider',
+
+        init: function init() {
+            var self  = this;
+            var areaBroker = this.getAreaBroker();
+
+            this.button = areaBroker.getToolbox().createEntry({
+                control: 'hider',
+                text: 'Hider',
+                title: 'Hide the item',
+                icon: 'eye-slash'
+            });
+            this.button.on('click', function(e) {
+                e.preventDefault();
+                areaBroker.getContentArea().toggle();
+            });
+
+            this.getTestRunner()
+                .on('enabletools renderitem', function (){
+                    self.enable();
+                })
+                .on('disabletools unloaditem', function (){
+                    self.disable();
+                });
+        },
+
+        render: function render() {},
+
+        destroy: function destroy() {},
+
+        enable: function enable() {
+            this.button.enable();
+        },
+
+        disable: function disable() {
+            this.button.disable();
+        },
+
+        show: function show() {},
+
+        hide: function hide() {}
+    });
+});
+```
+We have implemented the `enable` and `disable` method, retrieved the current test runner instance using `this.getTestRunner()` and listen some events triggered by the test runner :
+ - `renderitem` when an item is rendered to the test taker
+ - `unloaditem` when an item is removed from the screen
+ - `enabletools` and `disabletools` are used to control whether all tools should be available, for example, when displaying a modal popup the tools should be _disabled_.
+
+ The test runner provides much more useful events for the plugins, like
+ - `loaditem` when the item data is retrieved (but the item isn't yet displayed)
+ - `move`  or `skip` , so you know when the test taker move forward, or backward.
+ - `exit` or  `timeout` also pretty useful
+
+The complete list of events is available in the [test runner documentation](https://hub.taocloud.org/techdocs/tao-testqti/test-runner#events).
+
+For your information, since we can retrieve an instance of the test runner, we have access to the test and item data and metadata, by using the methods  :
+ - `this.getTestRunner().getTestData()`
+ - `this.getTestRunner().getTestContext()`
+ - `this.getTestRunner().getTestMap()`
+ - `this.getTestRunner().getItem()`
+
+
+Let's try now, we have a functional plugin !
+
+![hider plugin](resources/test-runner-plugins/hider-plugin.png)
+
+![hider plugin](resources/test-runner-plugins/hider-plugin-hidden.png)
+
+### Keep the room clean
+
+But before leaving, since we are conscientious people, we need to handle the plugin destroy correctly by removing event handlers :
```

Comment:
either 'we need to destroy the plugin' or we need to handle the plugin destruction